promql/engine.go

```diff
@@ -839,6 +869,21 @@ func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ..
 	return mat
 }
 
+// subqIntoMatrixSelector evaluates given SubqueryExpr and returns an equivalent
+// evaluated MatrixSelector in its place. Note that the Name and LabelMatchers are not set.
+func (ev *evaluator) subqIntoMatrixSelector(subq *SubqueryExpr) *MatrixSelector {
```

Comment:
I think a better name would be `evalSubquery`. From the current name it is not obvious that you are actually evaluating the subquery, and that is arguably the most important part of what you're doing here, not merely mapping the range and offset into an equivalent `MatrixSelector`.